vm-mock/vm.js

```diff
@@ -414,11 +414,20 @@ export default function createMockedABI(
 
       assembly_script_validate_address(addressPtr) {
         const address = ptrToString(addressPtr);
+        const prefix = 'A';
+        const userOrSC = ['U', 'S'];
+        const minLength = 40;
+        const maxLength = 60;
 
-        return (
-          stringToByteArray(address).length === addressBytesLength &&
-          (address.startsWith('AU') || address.startsWith('AS'))
-        );
+        if (address.length < minLength && address.length > maxLength) {
+          return false;
+        }
+
+        if (address.startsWith(prefix) && userOrSC.includes(address[1])) {
+          return true;
+        }
+
+        return false;
```

Comment:
You can simplify the code like this

```suggestion
        return address.startsWith(prefix) && userOrSC.includes(address[1])
```

In addition and no-blocking, I like when the condition is self-explaining, so that:
- it improves readeability
- you can unit test each condition

```ts
  return isRightLength(address) && isRightPrefix(address)
```

```ts
function isRigthLength(address) {
  return address.length >= minLength || address.length <= maxLength
}

function isRightPrefix(address) {
  return address.startsWith(prefix) && userOrSC.includes(address[1]))
}
```

then in the following PR, you will just change isRigthLength implementation


```ts
function isRigthLength(address) {
  return base58 decode is valid...
}
```

Reply:
great thanks! I'll change it

Reply:
Done